File: scala/RegEx.scala
import scala.util.matching.Regex

object RegEx {
  def main(args: Array[String]) {
    val pattern = new Regex("(S|s)cala")
    val str = "Scala is scalable and cool"
    println((pattern findAllIn str).mkString(",")) // Scala,scala
    println(pattern replaceFirstIn(str, "Java")) // Java is scalable and cool

    val p = "Sleep ([0-9]*)$".r
    "Sleep 100" match {
      case p(n) => println(n.toInt) // prints 100
      case _ => println("no match")
    }

    val simple = "^[/]?(__.*/)?foo/.*".r
    args(0) match {
      case simple(_) => println("match")
      case _ => println("unable to find root folder in: " + args(0) )
    }
  }
}

File: java/Regex.java
import java.util.regex.*;

class Regex {
  public static void main(String args[]) {
/*
    Pattern p = Pattern.compile(args[0]);
    Matcher m = p.matcher(args[1]);
    System.out.println(String.format("regex %s, input %s, matches %s", args[0], args[1], m.matches()));
*/

    Pattern p = Pattern.compile("^.*[0-9]*.*$");
    Matcher m = p.matcher("abc123xyz456");
    System.out.println(m.matches()); // prints true

    p = Pattern.compile("-?[0-9]+");
    m = p.matcher("abc123xyz456xxx");

    while (m.find()) {
      System.out.println(m.group());
    }
  }
}
